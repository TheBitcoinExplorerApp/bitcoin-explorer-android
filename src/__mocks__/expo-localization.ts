const getLocales = () => [
  { country: "pt", languageTag: "pt-BR", languageCode: "pt", isRTL: false },
];

export default getLocales;
