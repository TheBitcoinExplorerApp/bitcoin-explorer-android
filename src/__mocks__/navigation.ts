const mockNavigation = jest.fn();

jest.mock("@react-navigation/native", () => {
  const actualNavigation = jest.requireActual("@react-navigation/native");
  return {
    ...actualNavigation,
    useNavigation: () => ({
      navigate: mockNavigation,
    }),
  };
});
