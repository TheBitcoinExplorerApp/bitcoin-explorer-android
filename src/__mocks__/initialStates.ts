import { BlockType } from "src/components/BlocksInfo/components/Block/type";
import { AddressInfoType, TransactionType } from "src/components/Modal/types";

const timeAgoInitialState = {
  hour: 0,
  minutes: 0,
  day: 0,
  month: 0,
  year: 0,
};

export const initialStateBlocks: BlockType[] = [
  {
    blockHash: "",
    blockHeight: 1,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
    extras: {
      medianFee: 1,
      pool: {
        name: " ",
      },
    },
  },
  {
    blockHash: "",
    blockHeight: 2,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
    extras: {
      medianFee: 1,
      pool: {
        name: " ",
      },
    },
  },
  {
    blockHash: "",
    blockHeight: 3,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
    extras: {
      medianFee: 1,
      pool: {
        name: " ",
      },
    },
  },
  {
    blockHash: "",
    blockHeight: 4,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,

    extras: {
      medianFee: 1,
      pool: {
        name: " ",
      },
    },
  },
];

export const initialStateTransaction: TransactionType = {
  transactionId: " ",
  size: 0,
  fee: 0,
  inputTransactions: [
    {
      prevout: {
        scriptpubkey_address: " ",
        value: 0,
      },
    },
  ],
  outputTransactions: [
    {
      scriptpubkey_address: " ",
      value: 0,
    },
  ],
  statusTransaction: undefined,
};

export const initialStateFormattedTransactions = [
  {
    transactionId: "0",
    value: 0,
    fee: 0,
  },
  {
    transactionId: "1",
    value: 0,
    fee: 0,
  },
];

export const initialStateAddress: AddressInfoType = {
  address: "",
  balance: 0,
  totalAmountReceived: 0,
  totalAmountSent: 0,
};
