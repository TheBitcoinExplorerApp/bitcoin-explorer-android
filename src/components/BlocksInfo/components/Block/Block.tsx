/* eslint-disable import/no-extraneous-dependencies */
import { View, Text, StyleSheet, TouchableOpacity } from "react-native";
import { useState } from "react";
import Modal from "src/components/Modal/Modal";
import { initialStateBlocks } from "src/mocks/initialStates";
import ModalServices from "src/services/ModalServices";
import { TransactionType } from "src/components/Modal/types";
import useAppDataStore from "src/context/DataProvider";
import { BlockProps, BlockType } from "./type";

const styles = StyleSheet.create({
  blocksContainer: {
    flexDirection: "row",
    justifyContent: "space-between",
    flexWrap: "wrap",
    gap: 18,
  },
  blockContainer: {
    alignItems: "center",
    justifyContent: "center",
    width: 160,
    height: 110,
    backgroundColor: "#1d2133",
    borderRadius: 7,
  },
  textPrimary: {
    color: "#DF7800",
    fontSize: 15,
  },
  text: {
    color: "#FFF",
    fontSize: 12,
  },
});

export default function Block(props: BlockProps) {
  const { blocks } = props;
  const { i18n } = useAppDataStore();

  const [modalVisible, setModalVisible] = useState(false);
  const [blockData, setBlockData] = useState<BlockType>(initialStateBlocks[0]);
  const [blockTransactions, setBlockTransactions] = useState<TransactionType[]>(
    [],
  );
  const getTransactionsFromBlock = async (blockHash: string) => {
    const data = await ModalServices.getBlockTransactions(blockHash);

    setBlockTransactions(data);
  };

  return (
    <View style={styles.blocksContainer}>
      {blocks.map((block) => {
        return (
          <View key={block.blockHeight}>
            <TouchableOpacity
              onPress={() => {
                getTransactionsFromBlock(block.blockHash);
                setBlockData(block);
                setModalVisible(true);
              }}
            >
              <View
                key={block.blockHeight}
                style={{ ...styles.blockContainer }}
              >
                <Text style={styles.textPrimary}>{block.blockHeight}</Text>
                <Text style={styles.text}>{block.satPerVbyte} sat/vB</Text>
                <Text style={styles.text}>{block.size} MB</Text>
                <Text style={styles.text}>
                  {block.transactions} {i18n.t("transactions").toLowerCase()}
                </Text>
                <Text style={styles.text}>
                  {block.timeAgo.hour}:{block.timeAgo.minutes}
                </Text>
              </View>
            </TouchableOpacity>
          </View>
        );
      })}

      <Modal
        isVisible={modalVisible}
        handleModalClose={() => {
          setModalVisible(false);
          setBlockTransactions([]);
        }}
        modalType="Block"
        blockHash={blockData.blockHash}
        blockHeight={blockData.blockHeight}
        satPerVbyte={blockData.satPerVbyte}
        size={blockData.size}
        timeAgo={blockData.timeAgo}
        transactions={blockData.transactions}
        extras={blockData.extras}
        blockTransactions={blockTransactions}
      />
    </View>
  );
}
