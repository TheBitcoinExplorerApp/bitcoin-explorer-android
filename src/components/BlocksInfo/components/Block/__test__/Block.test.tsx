/* eslint-disable import/no-extraneous-dependencies */
import renderer from "react-test-renderer";
import { initialStateBlocks } from "src/__mocks__/initialStates";
import Block from "../Block";

describe("Block", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Block blocks={initialStateBlocks} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
