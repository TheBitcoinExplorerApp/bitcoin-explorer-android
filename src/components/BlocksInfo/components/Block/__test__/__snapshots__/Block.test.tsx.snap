// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": 18,
      "justifyContent": "space-between",
    }
  }
>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#1d2133",
            "borderRadius": 7,
            "height": 110,
            "justifyContent": "center",
            "width": 160,
          }
        }
      >
        <Text
          style={
            {
              "color": "#DF7800",
              "fontSize": 15,
            }
          }
        >
          1
        </Text>
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 12,
            }
          }
        >
          1
           sat/vB
        </Text>
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 12,
            }
          }
        >
          1
           MB
        </Text>
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 12,
            }
          }
        >
          1
           
          transações
        </Text>
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 12,
            }
          }
        >
          0
          :
          0
        </Text>
      </View>
    </View>
  </View>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#1d2133",
            "borderRadius": 7,
            "height": 110,
            "justifyContent": "center",
            "width": 160,
          }
        }
      >
        <Text
          style={
            {
              "color": "#DF7800",
              "fontSize": 15,
            }
          }
        >
          2
        </Text>
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 12,
            }
          }
        >
          1
           sat/vB
        </Text>
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 12,
            }
          }
        >
          1
           MB
        </Text>
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 12,
            }
          }
        >
          1
           
          transações
        </Text>
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 12,
            }
          }
        >
          0
          :
          0
        </Text>
      </View>
    </View>
  </View>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#1d2133",
            "borderRadius": 7,
            "height": 110,
            "justifyContent": "center",
            "width": 160,
          }
        }
      >
        <Text
          style={
            {
              "color": "#DF7800",
              "fontSize": 15,
            }
          }
        >
          3
        </Text>
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 12,
            }
          }
        >
          1
           sat/vB
        </Text>
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 12,
            }
          }
        >
          1
           MB
        </Text>
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 12,
            }
          }
        >
          1
           
          transações
        </Text>
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 12,
            }
          }
        >
          0
          :
          0
        </Text>
      </View>
    </View>
  </View>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#1d2133",
            "borderRadius": 7,
            "height": 110,
            "justifyContent": "center",
            "width": 160,
          }
        }
      >
        <Text
          style={
            {
              "color": "#DF7800",
              "fontSize": 15,
            }
          }
        >
          4
        </Text>
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 12,
            }
          }
        >
          1
           sat/vB
        </Text>
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 12,
            }
          }
        >
          1
           MB
        </Text>
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 12,
            }
          }
        >
          1
           
          transações
        </Text>
        <Text
          style={
            {
              "color": "#FFF",
              "fontSize": 12,
            }
          }
        >
          0
          :
          0
        </Text>
      </View>
    </View>
  </View>
</View>
`;
